fuel-block-producer/src/adapters.rs

```diff
@@ -0,0 +1,38 @@
+use crate::ports::TxPool;
+use fuel_core_interfaces::{
+    common::fuel_tx::{
+        CheckedTransaction,
+        ConsensusParameters,
+    },
+    model::BlockHeight,
+    txpool::Sender,
+};
+use std::sync::Arc;
+
+pub struct TxPoolAdapter {
+    pub sender: Sender,
+    pub consensus_params: ConsensusParameters,
+}
+
+#[async_trait::async_trait]
+impl TxPool for TxPoolAdapter {
+    async fn get_includable_txs(
+        &self,
+        block_height: BlockHeight,
+    ) -> anyhow::Result<Vec<Arc<CheckedTransaction>>> {
+        let includable_txs = self.sender.includable().await?;
```

Comment:
Maybe for production better to extract all transactions and move them to the block producer? BP will return them to the TX pool if the block can't be created.

Why do we want to duplicate transactions in the TX pool? Only for the GraphQL cases?

Reply:
Maybe it would be simpler to only deal with the removal of txs from the block importer/committer, that way txs are only removed once the block is actually applied? Since in PoS mode, we might not know right away whether the block is good and then we need come up with a new way to re-include the txs from a failed round.

Less copying is generally a good thing, but we'd also have to factor in the cost of re-insertion of txs that didn't fit in the block. Let's say our txpool is holding more than 100 blocks worth of txs, this repeated txpool dumping and reinsertion may be higher than just copying the txs we selected for the block.

Reply:
We can move the selection strategy into the tx pool. In that case, we can call `txpool.includable(MAX_GAS)`, it will return only transactions that cumulative gas is less than `MAX_GAS`.

TXs can be sorted in the `TxPool` by the gas, and selecting the last `X` transactions is easy. Insertion into this vector will be for  `log(n)` with a binary search approach.

Reply:
That might be a good idea. Maybe the txpool could even store "next block tx set" separately to make it even faster. I'm not sure this optimization is necessary though.

Reply:
This does make more sense than using this heavy "fetch all potentially incluable txs and then filter them" design IMO.

> TXs can be sorted in the TxPool by the gas, and selecting the last X transactions is easy. 

If you take a look at the current txpool impl you'll see this is already done :)

I'll update the txpool port for block producer to provide a max_gas to the includable API. For now can move the selection logic into the TxPool adapter, and then make a separate PR to update the txpool API and remove the need for the adapter to do any advanced logic.

Reply:
moved transaction selection to the adapter and updated the local txpool interface to include max_gas.